### PR TITLE
Apply black style to test_Restriction.py, version 19.10b0.

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #2552 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This PR applies black style to test_Restriction.py. few changes but should be fine to merge.

### CODE REVIEW

File: Tests/test_Restriction.py [hunk 1]
```diff
@@ -8,8 +8,23 @@
...
+    SmaI,
+    MluCI,
+    McrI,
+    NdeI,
+    BsmBI,
+    AanI,
+    EarI,
+    SnaI,
+    SphI,
+)
```
Comment (peterjc) [inline Tests/test_Restriction.py:27]: My personal preference would be:

```python
from Bio.Restriction import Acc65I
...
from Bio.Restriction import SphI
```

See discussion on #2828 - we have yet to agree any particular style, so this is fine for now.

File: Tests/test_Restriction.py [hunk 2]
```diff
@@ -38,13 +53,12 @@ def test_non_iupac_letters(self):
...
     def test_formatted_seq(self):
         """Test several methods of FormattedSeq."""
-        self.assertEqual(str(FormattedSeq(Seq("GATC"))),
-                         "FormattedSeq(Seq('GATC'), linear=True)")
-        self.assertFalse(FormattedSeq(Seq("GATC")) ==
-                         FormattedSeq(Seq("TAGC")))
+        self.assertEqual(
+            str(FormattedSeq(Seq("GATC"))), "FormattedSeq(Seq('GATC'), linear=True)"
+        )
+        self.assertFalse(FormattedSeq(Seq("GATC")) == FormattedSeq(Seq("TAGC")))
```
Comment (MarkusPiotrowski) [inline Tests/test_Restriction.py:59]: @svalqui Sorry for this request, although not black-related but still style-related: Can you change all `assertFalse(a==b)` and `assertTrue(a==b)` into `assertNotEqual(a, b)` and `assertEqual(a, b)`?

File: Tests/test_Restriction.py (hunk 1 above)
Comment (MarkusPiotrowski) [reply Tests/test_Restriction.py:27]: My personal preference would be to let it stand as it was, but let black do what black wants to do here.

File: Tests/test_Restriction.py (hunk 2 above)
Comment (MarkusPiotrowski) [reply Tests/test_Restriction.py:59]: Not only here but in the whole file?

File: Tests/test_Restriction.py (hunk 1 above)
Comment (peterjc) [reply Tests/test_Restriction.py:27]: Yes, leave it as is for now - we can settle any preferred import style later on the linked issue.

File: Tests/test_Restriction.py (hunk 2 above)
Comment (peterjc) [reply Tests/test_Restriction.py:59]: Logged #2835 for the wider issue, but happy for you to fix this file as part of this pull request, or leave it for later.

Comment (svalqui) [reply Tests/test_Restriction.py]: updated, but left one that doesn't pass the test.

```
    self.assertNotEqual(Acc65I, Asp718I)
AssertionError: Acc65I == Asp718I

```
I think is mean to use assertNotEqual.

Comment (MarkusPiotrowski) [reply Tests/test_Restriction.py]: This is a special thing within `Bio.Restriction`, `__eq__` (`==`) and `__ne__` (`!=`) have been overriden with different meanings. In this case Acc65I and Asp718I are neither equal (they are not the same enzyme) nor not equal (they are isoschizomers, doing exactly the same thing). 
Thus `assertFalse` is the only method which allows to test both. OK.

Comment (peterjc) [reply Tests/test_Restriction.py]: I'd forgotten about that quirk. Best or at least standard practise is ``__eq__`` and ``__ne__`` give the negation of each other, but as here that isn't required.

**Comment by peterjc** [comment]
Thank you both

File: Tests/test_Restriction.py (hunk 2 above)
Comment (ajsmith) [reply Tests/test_Restriction.py]: @peterjc @MarkusPiotrowski Perhaps I'm new here, but is there any reason why the `__ne__` operator is used to check for isoschizomer-ness? It doesn't seem very convenient if it confuses the linter and tests (and potentially humans too). When I first saw this behavior in the tests, I thought it was a bug, but then realized it was by design.

Comment (peterjc) [reply Tests/test_Restriction.py]: I would agree in hindsight this was a poor design choice. We have been looking at replacing the entirety of the old restriction digest module anyway.

Comment (ajsmith) [reply Tests/test_Restriction.py]: @peterjc Thank you for the clarification, and I look forward to seeing the new design!

**Pull request merged**